### ISSUE TITLE
Not working with Bootstrap 4.2?

### ISSUE THREAD

**Issue by fijas** [scttcper/ngx-toastr#602]
Seems like this component is not working when using the newly released Bootstrap 4.2. A quick investigation shows that BS4.2 is using the .toast class which is hiding ngx-toastr toasts. Any possibility of namespacing the toast classes of ngx-toastr? Perhaps with toastr-* or something similar?

**Comment by scttcper** [comment]
yeah theres some conflicts. i recommend not importing bootstraps toast css for now. You can import only parts of bootstrap as in the demo here https://github.com/scttcper/ngx-toastr/blob/master/src/styles.scss#L1-L37

For a list of all imports see bootstrap itself - https://github.com/twbs/bootstrap/blob/v4-dev/scss/bootstrap.scss

**Comment by bpatrik** [comment]
It would be nice in the future, if the package would warn if an untested bootstrap is used.
Like: a warning during npm install. But the very best would be if a console.warning() would show an 'untested version' warning if the angular app is not in production mode. (enableProdMode() was not called)

**Comment by nikhilpatel7** [comment]
There is an issue with naming collision for the .toastr CSS class in bootstrap and ngx-toastr.
You can find the solution below till they fix it.
https://stackoverflow.com/questions/53989445/ngx-toastr-toast-not-showing-in-angular-7

**Comment by dstj** [comment]
> There is an issue with naming collision for the .toastr CSS class in bootstrap and ngx-toastr.
> You can find the solution below till they fix it.
> https://stackoverflow.com/questions/53989445/ngx-toastr-toast-not-showing-in-angular-7

That solution works in some cases, but when using `positionClass: 'toast-top-full-width'` for example, it doesn't do a good job.

**Comment by cmckni3** [comment]
ah yes. Bootstrap recently introduced the Toast component.

**Comment by mitschmidt** [comment]
Can confirm this issue, all toasts disappeared after updating bootstrap to `4.2` - thanks for the hint with the style sheets, that works for me as well.

**Comment by tbadlov** [comment]
This is due to conflicts with bootstrap 4.2.1 toast component. For now you can override by adding another style element to your project:

```scss
#toast-container > div {
  opacity:1;
}
```

**Comment by MikeOne** [comment]
Even though the recommended fix is workable - I would like to suggest to rename all ngx-toastr styles to reflect the namespace to prevent clashes like this.

**Comment by bpatrik** [comment]
I agree with @MikeOne, I find the recommended fix more a temporary solution / workaround then a real solution to the problem.

**Comment by Sef1995** [comment]
When i set the opacity to 1 I can see my toast but the background color of it is white instead of red.

**Comment by dalu** [comment]
there is no notyf.css in the ngx-toastr npm package

**Comment by yarrgh** [comment]
> When i set the opacity to 1 I can see my toast but the background color of it is white instead of red.

Are you by chance importing bootstrap into your styles.scss file and then importing toastr.css?

Doing that caused me to see the same issue as you, even though I imported toastr.css after all bootstrap imports. Inspecting the generated css puts bootstrap's `.toastr` class after all of the classes in toastr.css and overwrites the background color. 

To solve this I had to remove the import for toastr.css out of styles.scss and move it into the angular.json file, in the styles array after styles.scss.

```scss
"styles": ["src/styles.scss", "node_modules/ngx-toastr/toastr.css"],
```
